app/views/tasks/show.html.slim

```diff
@@ -478,13 +478,21 @@
             = button_tag class: 'btn btn-light dropdown-toggle', 'data-bs-toggle': 'dropdown' do
               = t('.button.add_to_collection')
             ul.scrollable.dropdown-menu[role="menu"]
-              - if !current_user.collections.empty?
-                - current_user.collections.reverse.each do |collection|
-                  li
-                    = link_to collection.title, add_to_collection_task_path(collection: collection), method: 'post', class: 'dropdown-item'
-              - else
+              - current_user.collections.reverse.each do |collection|
                 li
-                  = link_to t('.button.add_collection'), new_collection_path(current_user)
+                  - if collection.tasks.include?(@task)
+                    = link_to "<span class=\"collection-checkmark\">✓</span>#{collection.title}".html_safe, remove_task_ajax_collection_path(collection, task: @task), method: :patch, class: 'dropdown-item', data: { confirm: t('.remove_task_from_collection_warning') }
+                  - else
+                    = link_to "<span class=\"collection-checkmark\"></span>#{collection.title}".html_safe, add_to_collection_task_path(collection: collection), method: :post, class: 'dropdown-item'
+
+              li
+                hr.dropdown-divider
+              li
+                = form_with model: Collection.new, url: create_ajax_collections_path do |f|
+                  .create-collection.input-group
+                    = f.text_field :title, class: 'form-control', placeholder: t('.create_collection_placeholder')
+                    = f.hidden_field :task_ids, value: @task.id
```

Comment:
`ids` vs. `id` is confusing for me. But I don't know the domain.

Reply:
I understand the confusion, but I think it makes sense:
A `Collection` `has_many` `Tasks` and here we want to create a collection with one `Task` already added